utils/remote-ext-tests/replay-block/src/main.rs

```diff
@@ -0,0 +1,221 @@
+    Number(NumberFor<B>),
+}
+
+#[derive(Clone, Debug, Parser)]
+struct Opt {
+    /// The RPC url.
+    #[arg(
+		short,
+		long,
+		value_parser = parse::url,
+		default_value = "wss://archive-rpc.vara-network.io:443"
+	)]
+    uri: String,
+
+    /// The block hash or number we want to replay. If omitted, the latest finalized block is used.
+    /// The blockchain state at previous block with respect to this parameter will be scraped.
+    #[arg(
+		short,
+		long,
+		value_parser = parse::block,
+	)]
+    block: Option<BlockHashOrNumber<Block>>,
+
+    /// Pallet(s) to scrape. Comma-separated multiple items are also accepted.
+    /// If empty, entire chain state will be scraped.
+    #[arg(short, long, num_args = 1..)]
+    pallet: Vec<String>,
+
+    /// Fetch the child-keys as well.
+    ///
+    /// Default is `false`, if specific `--pallets` are specified, `true` otherwise. In other
+    /// words, if you scrape the whole state the child tree data is included out of the box.
+    /// Otherwise, it must be enabled explicitly using this flag.
+    #[arg(long)]
+    child_tree: bool,
+
+    /// Sets a custom logging filter. Syntax is `<target>=<level>`, e.g. -lsync=debug.
+    ///
+    /// Log levels (least to most verbose) are error, warn, info, debug, and trace.
+    /// By default, all targets log `info`. The global log level can be set with `-l<level>`.
+    #[arg(short = 'l', long, value_name = "LOG_PATTERN", num_args = 1..)]
+    pub log: Vec<String>,
+}
+
+impl Opt {
+    fn log_filters(&self) -> sc_cli::Result<String> {
+        Ok(self.log.join(","))
+    }
+}
+
+#[tokio::main]
+async fn main() -> sc_cli::Result<()> {
+    let options = Opt::parse();
+
+    let logger = LoggerBuilder::new(options.log_filters()?);
+    logger.init()?;
+
+    let ss58_prefix = match options.uri.contains("vara") {
+        true => VARA_SS58_PREFIX,
+        false => GEAR_SS58_PREFIX,
+    };
+    sp_core::crypto::set_default_ss58_version(ss58_prefix.try_into().unwrap());
+
+    // Initialize the RPC client.
+    // get the block number associated with this block.
+    let block_ws_uri = options.uri.clone();
+    let rpc = ws_client(&block_ws_uri).await?;
+
+    let block_hash_or_num = match options.block {
+        Some(b) => b,
+        None => {
+            let height = ChainApi::<
+                (),
+                <Block as BlockT>::Hash,
+                <Block as BlockT>::Header,
+                SignedBlock<Block>,
+            >::finalized_head(&rpc)
+            .await
+            .map_err(rpc_err_handler)?;
+
+            log::info!(
+                target: LOG_TARGET,
+                "Block is not provided, setting it to the latest finalized head: {:?}",
+                height
+            );
+
+            BlockHashOrNumber::Hash(height)
+        }
+    };
+
+    let (current_number, current_hash) = match block_hash_or_num {
+        BlockHashOrNumber::Number(n) => (n, block_number_to_hash::<Block>(&rpc, n).await?),
```

Comment:
I see that number is possible so state it in the docs please

Reply:
The `README.md` has the full description. It'a just the first part refers to the in-built `try-runtime` node subcommand (and how it can be used for the similar purpose).